libparsec/crates/platform_http_proxy/src/native/windows.rs

```diff
@@ -0,0 +1,80 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use widestring::U16CStr;
+use windows_sys::Win32::{
+    Foundation::GetLastError,
+    Networking::WinHttp::{
+        WinHttpGetIEProxyConfigForCurrentUser, WINHTTP_CURRENT_USER_IE_PROXY_CONFIG,
+    },
+    System::Memory::GlobalFree,
+};
+
+use libparsec_types::anyhow;
+
+use crate::ProxyConfig;
+
+impl ProxyConfig {
+    pub(crate) fn with_register(self) -> anyhow::Result<Self> {
+        let mut cfg = self;
+
+        let mut proxy_config = WINHTTP_CURRENT_USER_IE_PROXY_CONFIG {
+            fAutoDetect: 0,
+            lpszAutoConfigUrl: std::ptr::null_mut(),
+            lpszProxy: std::ptr::null_mut(),
+            lpszProxyBypass: std::ptr::null_mut(),
+        };
+
+        // Safety: Nothing unsafe here
+        unsafe {
+            if WinHttpGetIEProxyConfigForCurrentUser(&mut proxy_config) == 0 {
+                return Err(anyhow::anyhow!(
+                    "WinHttpGetProxyForUrl failed with code: {}",
+                    GetLastError()
+                ));
+            }
+        }
+
+        // Proxy Auto Config enabled
+        if !proxy_config.lpszAutoConfigUrl.is_null() {
+            // Safety: According to Windows doc, we should free it
+            // https://learn.microsoft.com/en-us/windows/win32/api/winhttp/nf-winhttp-winhttpgetieproxyconfigforcurrentuser#remarks
```

Comment:
That is not a reason why it's safe, what's important here is we don't free something that we don't own or something that will be used after (Use after free vuln).

Reply:
It's safe because the `Windows API` allocated the data